### PR TITLE
Shrd vignette test

### LINKED ISSUE #72: SHRD - Refactor PrepForLevelsInitialization to not require intermediate file
We currently create an intermediate file unnecessarily, which is created and read by `PrepForLevelsInitialization `. This should be refactored to be entirely in-memory.

### PR DESCRIPTION
**Pull request by brandenjlynch**
Resolves #72, #45, and #76.

### CODE REVIEW

File: vignettes/4_shallowhrd-vignette.Rmd [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+Annotation: 
+```
+
+---
+
+### III - Extract data and run shallowHRD
+
+---
+
+We can run shallowHRD directly on the QDNAseq object we just read in, using the `RunShallowHRDFromQDNA()` function, or on a table of segments, using the `RunShallowHRD()` function. Since estimating the threshold used to merge segments requires us to take 100,000 samples by default, running a single sample can take a few minutes. In the interest of time, we will extract a single sample from the QDNAseq object and demonstrate running shallowHRD on a table of segments. To run it for all the samples in the object, we could just call `RunShallowHRDFromQDNA()` directly on the QDNAseq object, with most arguments remaining the same as those we'll provide to `RunShallowHRD()`.
```
Comment (genomaxx) [inline vignettes/4_shallowhrd-vignette.Rmd:70]: The second sentence of this paragraph is a bit confusing. 100,000 samples of what? Are these 100,000 samples the the same as the 'sample' written later in the sentence?

File: vignettes/4_shallowhrd-vignette.Rmd [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+```{r, out.width = '60%', echo = FALSE}
+knitr::include_graphics("images/shallowhrd_vignette_1.png")
+```
+
+In this plot we can see out usual rCN profile, log-transformed and centered about 0. The thick green bars indicate the original segmentation. To visualize the process of segment merging over two passes, the medium thickness aquamarine bars illustrate the segments after merging with the first threshold, and the thinnest bright green bars show the final segmentation, after merging using the threshold estimated during the second pass. We can see that many segments are merged in the first pass, with only a few undergoing additional merging or un-merging during the second pass, such as those near the middle of chromosome 7.
```
Comment (genomaxx) [inline vignettes/4_shallowhrd-vignette.Rmd:175]: our*

File: vignettes/4_shallowhrd-vignette.Rmd [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
...d CN data. Because of noise that is inherent to the sampling process, which varies as a function of read depth, among other factors like GC-content, mappability, and artifacts derived from fixation, segmentation is necessarily imperfect. ShallowHRD therefore seeks to merge adjacent segments which differ in CN by less than a threshold in order to better identify LGAs. In the first pass, kernel density estimation (KDE) is used to find a local minimum in the segment-wise CN distance -- that is, the CN distance between adjacent segments. Adjacent segments with CN distance less that the identified threshold are merged and their segmented CN values are recaclucated accordingly.  After merging segments in the first pass, a second estimate of the optimal merging threshold is determined in the same fashion, this time using the updated segmented CN values produced by the first pass. This provides a refined estimate of the merging threshold, which is then used to merge the original segments, yielding the final set of merged segments. Finally, the LGAs in the sample are counted, and the HRD status is determined; samples with 20 or more LGAs are considered HRD, with the remainder being non-HRD.
```
Comment (genomaxx) [inline vignettes/4_shallowhrd-vignette.Rmd:28]: recalculated*

**Review by genomaxx — COMMENTED** [review]
Looks good overall, but a few typos and then a sentence that needs to be re-worded.

File: vignettes/4_shallowhrd-vignette.Rmd (hunk 1 above)
Comment (brandenjlynch) [reply vignettes/4_shallowhrd-vignette.Rmd]: Reworded this.

File: vignettes/4_shallowhrd-vignette.Rmd (hunk 2 above)
Comment (brandenjlynch) [reply vignettes/4_shallowhrd-vignette.Rmd]: Fixed.

File: vignettes/4_shallowhrd-vignette.Rmd (hunk 3 above)
Comment (brandenjlynch) [reply vignettes/4_shallowhrd-vignette.Rmd]: Fixed.

**Review by genomaxx — APPROVED** [review]
Looks good.

**Pull request merged**